tests/components/VerticalResults.test.tsx

```diff
@@ -72,152 +29,50 @@ const mockedActions = {
-    const paginationNavEl = screen.getByRole('navigation', { name: 'Pagination' });
-    expect(paginationNavEl).toBeDefined();
-    const totalPaginationButtons = within(paginationNavEl).queryAllByRole('button').length;
-    const numIconNavButtons = 2;
-    const numLabelNavButtons = mockedVerticalSearchState.resultsCount;
-    expect(totalPaginationButtons).toEqual(numIconNavButtons + numLabelNavButtons);
-  });
+it('displays results', () => {
+  const verticalResultsProps: VerticalResultsProps = {
+    CardComponent: StandardCard
+  };
+  render(<VerticalResults {...verticalResultsProps} />);
+  mockedState.vertical?.results?.forEach(result =>
+    expect(screen.getByText(result.rawData.name as string)).toBeDefined()
+  );
+});
 
-  it('Pagination component is displayed with ellipses label', () => {
-    const verticalResultsProps: VerticalResultsProps = {
-      CardComponent: StandardCard,
-      allowPagination: true,
-      customCssClasses: {
-        leftIconContainer: 'leftNavButton',
-        rightIconContainer: 'rightNavButton'
-      }
-    };
-    const mockedVerticalSearchState: VerticalSearchState = {
-      results: [mockedResults[0]],
-      resultsCount: 5,
-      verticalKey: 'vertical',
-      limit: 1,
-      offset: 0
-    };
-    mockVerticalSearchState(mockedVerticalSearchState);
-    render(<VerticalResults {...verticalResultsProps} />);
-    const paginationNavEl = screen.getByRole('navigation', { name: 'Pagination' });
-    expect(paginationNavEl).toBeDefined();
-    const paginationButtons = within(paginationNavEl).queryAllByRole('button');
-    const numIconNavButtons = 2;
-    const numLabelNavButtons = 3;
-    // expected pagination layout with n results: [<] [1] [2] [...] [n] [>]
-    expect(paginationButtons.length).toEqual(numIconNavButtons + numLabelNavButtons);
-    expect(paginationButtons[0].classList.contains('leftNavButton')).toBeTruthy();
-    expect(paginationButtons[1].textContent).toEqual('1');
-    expect(paginationButtons[2].textContent).toEqual('2');
-    expect(paginationButtons[3].textContent).toEqual(`${mockedVerticalSearchState.resultsCount}`);
-    expect(paginationButtons[4].classList.contains('rightNavButton')).toBeTruthy();
-    expect(screen.getByText('...')).toBeDefined();
+it('displays all results when there are no search results', () => {
+  const verticalResultsProps: VerticalResultsProps = {
+    CardComponent: StandardCard,
+    displayAllOnNoResults: true
+  };
+  const mockedNoResults = {
+    allResultsForVertical: {
+      facets: [],
+      results: mockedVerticalResults,
+      resultsCount: mockedVerticalResults.length
+    },
+    alternativeVerticals: []
+  };
+  mockVerticalSearchState({
+    noResults: mockedNoResults
   });
+  render(<VerticalResults {...verticalResultsProps} />);
+  mockedNoResults.allResultsForVertical.results.forEach(result =>
+    expect(screen.getByText(result.rawData.name as string)).toBeDefined()
```

Comment:
is this assert necessary?

Reply:
looks like it was here before but if we can remove it that'd be good!

Reply:
(also the other one)

Reply:
When it's removed, ts complains. Since I think `getByText` is an overloaded function and `result.rawData.name` ends up being type `unknown`, so it doesn't really know what to do unless we specify the type of the input.

Reply:
oh I see makes sense! here result.rawData is a map of strings to unknown values which is something the generate-types CLI command is going to help with